plugin/pkg/shoot/validator/admission.go

```diff
@@ -508,11 +508,18 @@ func validateKubernetesVersionConstraints(constraints []core.ExpirableVersion, s
 	validValues := []string{}
 	var latestVersion *semver.Version
 	for _, versionConstraint := range constraints {
-		if versionConstraint.ExpirationDate != nil && versionConstraint.ExpirationDate.Time.Before(time.Now()) {
+		if versionConstraint.ExpirationDate != nil && versionConstraint.ExpirationDate.Time.UTC().Before(time.Now().UTC()) {
 			continue
 		}
 
-		validValues = append(validValues, versionConstraint.Version)
+		if !getLatestPatchVersion {
+			validValues = append(validValues, versionConstraint.Version)
+		} else {
+			// ignore preview versions for defaulting
+			if versionConstraint.Classification != nil && *versionConstraint.Classification == core.ClassificationPreview {
+				continue
+			}
+		}
 
 		if versionConstraint.Version == shootVersion {
```

Comment:
Can you move this if block up, before the previous one? If I see this correctly, the Shoot won't be able to use a Preview version otherwise...

Reply:
The Shoot can use a preview version. I wrote a new test for it in the admission_test.
Preview versions are only filtered here when the Shoot only provides a major.minor version like 1.6

Reply:
Oh yes, sorry.